app/routes/participate.js

```diff
@@ -32,10 +32,12 @@ export default Ember.Route.extend(WarnOnExitRouteMixin, {
   },
   model(params) {
     return new Ember.RSVP.Promise((resolve, reject) => {
+      var _this = this;
       this._getExperiment(params).then((experiment) => {
         this._getSession(params, experiment).then((session) => {
           this.set('_experiment', experiment);
           session.set('experimentVersion', '');
+          session.set('locale', _this.controllerFor('index').get('locale'));
```

Comment:
If we use this approach is would be good to add a `beforeModel` hook to be sure that the user has transitioned from the login page (otherwise they should be redirected to the login page)